File: grpc-quickstart/src/main/java/com/kdnakt/quarkus/grpc/GreetingResource.java
package com.kdnakt.quarkus.grpc;

import io.quarkus.example.GreeterGrpc;
import io.quarkus.example.HelloRequest;
import io.quarkus.grpc.runtime.annotations.GrpcService;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/greeting")
public class GreetingResource {

    @Inject
    @GrpcService("hello")
    GreeterGrpc.GreeterBlockingStub client;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return "hello";
    }

    @GET
    @Path("/{name}/{age}")
    public String hello(@PathParam("name") String name,
            @PathParam("age") int age) {
        return client.sayHello(
                HelloRequest.newBuilder().setName(name).setAge(age).build()
                ).getMessage();
    }

}

File: rest-kotlin-quickstart/src/main/kotlin/com/kdnakt/quarkus/rest/GreetingResource.kt
package com.kdnakt.quarkus.rest

import javax.inject.Inject
import javax.enterprise.inject.Default
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/")
class GreetingResource {

    @Inject
    @field: Default
    lateinit var service: GreetingService

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    @Path("/greeting/{name}")
    fun greeting(@PathParam("name") name: String): String {
        return service.greeting(name)
    }
}